components/centraldashboard-angular/frontend/src/app/pages/iframe-wrapper/iframe-wrapper.component.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  OnDestroy,
+  ViewChild,
+} from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-iframe-wrapper',
+  templateUrl: './iframe-wrapper.component.html',
+  styleUrls: ['./iframe-wrapper.component.scss'],
+})
+export class IframeWrapperComponent implements AfterViewInit, OnDestroy {
+  @ViewChild('iframe') iframe: ElementRef<HTMLIFrameElement>;
+
+  public prvSrcPath: string;
+  get srcPath(): string {
+    return this.prvSrcPath;
+  }
+  set srcPath(src: string) {
+    /*
+     * The following hacky logic ensures that the Iframe will reload,
+     * even when it receives values for src that are the same with the
+     * one it already has. This is because, in order to avoid the iframe
+     * reloading constantly, its src is not updated on each navigation
+     * a user does.
+     */
+    if (window.location.origin) {
+      if (!this.prvSrcPath?.includes(window.location.origin)) {
+        src = window.location.origin + src;
+      }
+    }
+
+    this.prvSrcPath = src;
+    // Some KF distributions need a trailing slash "/" in order to resolve paths
+    this.prvSrcPath += this.prvSrcPath?.endsWith('/') ? '' : '/';
+  }
+  public iframeLocation: string | undefined = 'about:blank';
+  private urlSub: Subscription;
+  private interval: any;
+
+  constructor(private router: Router) {
+    this.urlSub = this.router.events.subscribe(event => {
+      if (!(event instanceof NavigationEnd)) {
+        return;
+      }
+
+      const iframeWindow = this.iframe?.nativeElement?.contentWindow;
+      let iframeUrl = iframeWindow?.location.pathname;
+      if (iframeUrl) {
+        // Include URL's query parameters
+        iframeUrl += iframeWindow?.location.search;
+      }
+
+      if (!this.equalUrlPaths(event.url, iframeUrl)) {
+        this.srcPath = event.url;
+      }
+    });
+  }
+
+  equalUrlPaths(firstUrl: string, secondUrl: string | undefined) {
+    if (!firstUrl && !secondUrl) {
```

Comment:
Let's add a comment here to explain why we need this function, which is to handle the sometimes missing `/` from some urls